sdk/storage/inc/datalake/path_client.hpp

```diff
@@ -0,0 +1,285 @@
+     * @brief Create from connection string
+     * @param connectionString Azure Storage connection string.
+     * @param fileSystemName The name of a file system.
+     * @param path The path of a resource within the file system.
+     * @param options Optional parameters used to initialize the client.
+     * @return PathClient
+     */
+    static PathClient CreateFromConnectionString(
+        const std::string& connectionString,
+        const std::string& fileSystemName,
+        const std::string& path,
+        const PathClientOptions& options = PathClientOptions());
+
+    /**
+     * @brief Shared key authentication client.
+     * @param pathUri The URI of the file system this client's request targets.
+     * @param credential The shared key credential used to initialize the client.
+     * @param options Optional parameters used to initialize the client.
+     */
+    explicit PathClient(
+        const std::string& pathUri,
+        std::shared_ptr<SharedKeyCredential> credential,
+        const PathClientOptions& options = PathClientOptions());
+
+    /**
+     * @brief Bearer token authentication client.
+     * @param pathUri The URI of the file system this client's request targets.
+     * @param credential The token credential used to initialize the client.
+     * @param options Optional parameters used to initialize the client.
+     */
+    explicit PathClient(
+        const std::string& pathUri,
+        std::shared_ptr<TokenCredential> credential,
+        const PathClientOptions& options = PathClientOptions());
+
+    /**
+     * @brief Anonymous/SAS/customized pipeline auth.
+     * @param pathUri The URI of the file system this client's request targets.
+     * @param options Optional parameters used to initialize the client.
+     */
+    explicit PathClient(
+        const std::string& pathUri,
+        const PathClientOptions& options = PathClientOptions());
+
+    /**
+     * @brief Uploads data to be appended to a file. Data can only be appended to a file.
+     * @param data The data to be appended.
+     * @param position This parameter allows the caller to upload data in parallel and control
+     *                 the order in which it is appended to the file.
+     *                 The value must be the position where the data is to be appended.
+     *                 Uploaded data is not immediately flushed, or written, to the file. To flush,
+     *                 the previously uploaded data must be contiguous, the position parameter must
+     *                 be specified and equal to the length of the file after all data has been
+     *                 written, and there must not be a request entity body included with the
+     *                 request.
+     * @param options Optional parameters to append data to the resource the path points to.
+     * @return PathAppendDataResponse
+     */
+    PathAppendDataResponse AppendData(
+        std::vector<uint8_t> data,
+        int64_t position,
```

Comment:
`offset` or `position`, which is better?

Reply:
The official name is `position`REST.

Reply:
But this is convenience layer, not protocol layer.

Plus in .NET SDK, they call it `offset`.

Reply:
Why are you keep marking these unresolved conversation resolved?

Reply:
Changed to offset.